app/src/main/java/com/willowtree/vocable/splash/SplashViewModel.kt

```diff
@@ -20,8 +29,83 @@ class SplashViewModel : BaseViewModel() {
 
     private fun populateDatabase() {
         backgroundScope.launch {
+            val newCategoryId = UUID.randomUUID().toString()
+            val shouldUpdateName = presetsRepository.getPhrasesForCategory(PresetCategories.USER_FAVORITES.id)
+                .isNullOrEmpty()
+
+            moveMySayings(newCategoryId)
+
             presetsRepository.populateDatabase()
+
+            // we need to call this after populating the db so that the resource IDs are correct
+            updateMySayingsName(newCategoryId)
+
             liveExitSplash.postValue(true)
         }
     }
-}
\ No newline at end of file
+
+    private suspend fun updateMySayingsName(newCategoryId: String) {
+        val mySayingsCategory = presetsRepository.getCategoryById(newCategoryId)
+        // we need to check for null here, because if the category does not exist, the return value will be null
+        // regardless of the return type being non-nullable.
+        if (mySayingsCategory != null) {
+            mySayingsCategory.localizedName = mapOf(
+                Pair(
+                    Locale.getDefault().toString(),
+                    get<Context>().resources.getString(R.string.preset_user_favorites)
+                )
+            )
+
+            presetsRepository.updateCategory(mySayingsCategory)
+        }
+    }
+
+    private suspend fun moveMySayings(newCategoryId: String) {
+        // Get the old My Sayings category
+        val mySayingsCategory =
+            presetsRepository.getCategoryById(PresetCategories.USER_FAVORITES.id)
+
+        // if the user has My Sayings phrases, we need to migrate them to a custom category
+        if (presetsRepository.getPhrasesForCategory(PresetCategories.USER_FAVORITES.id)
+                .isNotEmpty()
+        ) {
+            val allCategories = presetsRepository.getAllCategories()
+
+            // Get the index of the first hidden category to find the sort order of new category
+            var firstHiddenIndex = allCategories.indexOfFirst { it.hidden }
```

Comment:
Could we also try to pull the sort order from the old My Sayings category first and then calculate it if it's null?

Reply:
I don't think we want to do that, because it's considered a custom category now, so it should be at the end, right?

Reply:
Ah fair point